src/jolt/gui/jolt_app.py

```diff
@@ -532,34 +740,77 @@ def on_voltage_button(self, event):
 
             if self.hv:
                 # write parameters to device
-                self.dev.set_voltage(self.voltage_gui)
+                self.dev.set_voltage_forced(self.voltage_gui)
+                if self.valid_calibration:
+                    self.set_matching_calibrated_offset()
                 self.spinctrl_voltage.SetForegroundColour(wx.Colour(wx.BLACK))
             else:
                 self.dev.set_voltage(0)
                 # light grey to show it's not actually set
                 self.spinctrl_voltage.SetForegroundColour((211, 211, 211))
-                self.spinctrl_voltage.SetValue(self.voltage_gui)
         self.refresh()
 
     def on_auto_bc(self, event):
         raise NotImplementedError()
         # disable gain/offset controls
 
+    def set_matching_calibrated_offset(self):
+        # Ignore temperature inaccuracies for now and just use the target to simplify things.
+        voltage_dists = {i: abs(r["voltage"] - self.voltage_gui) for i, r in enumerate(self.calibration_table)
+            if r["channel"] == STR2CHANNEL[self.channel]}
+
+        # Obtain the two closest calibration points for interpolation
+        idx_within_channel_low = argmin(voltage_dists.values())[0]
+        idx_within_channel_high = argmin(voltage_dists.values())[1]
+
+        match_idx_low = list(voltage_dists.keys())[idx_within_channel_low]
+        match_idx_high = list(voltage_dists.keys())[idx_within_channel_high]
+
+        matching_voltage_low = self.calibration_table[match_idx_low]["voltage_target"]
+        matching_voltage_high = self.calibration_table[match_idx_high]["voltage_target"]
+
+        matching_offset_low = self.calibration_table[match_idx_low]["fe_offset"]
+        matching_offset_high = self.calibration_table[match_idx_high]["fe_offset"]
+
+        # Interpolate to obtain the matching offset
+        slope, intercept = linear_regression([
+            (matching_voltage_low, matching_offset_low),
+            (matching_voltage_high, matching_offset_high),
+        ])
+
+        matching_offset = int(slope * self.voltage_gui + intercept)
+
+        # Clip
+        matching_offset = min(matching_offset, 1023)
+        matching_offset = max(matching_offset, 0)
```

Comment:
Typically we write such clipping in one line, like this:
```suggestion
        matching_offset = max(0, min(matching_offset, 1023))
```